src/twisted/test/test_process.py

```diff
@@ -2383,25 +2337,46 @@ class Win32UnicodeEnvironmentTests(unittest.TestCase):
     """
     Tests for Unicode environment on Windows
     """
-    goodKey = u'UNICODE'
-    goodValue = u'UNICODE'
+    goodKey = 'UNICODE'
+    goodKeyBytes = b'UNICODE'
+    goodValue = 'UNICODE'
+    goodValueBytes = b'UNICODE'
 
 
     def test_encodableUnicodeEnvironment(self):
         """
-        Test C{os.environ} (inherited by every subprocess on Windows) that
-        contains an ascii-encodable Unicode string. This is different from
-        passing Unicode environment explicitly to spawnProcess (which is not
-        supported on Python 2).
+        Test L{os.environ} (inherited by every subprocess on Windows) that
+        contains an ascii-encodable Unicode string.
         """
         os.environ[self.goodKey] = self.goodValue
         self.addCleanup(operator.delitem, os.environ, self.goodKey)
 
-        p = GetEnvironmentDictionary.run(reactor, [], properEnv)
-        def gotEnvironment(environ):
-            self.assertEqual(
-                environ[self.goodKey.encode('ascii')],
-                self.goodValue.encode('ascii'))
+        p = GetEnvironmentDictionary.run(reactor, [], os.environ)
+
+        def gotEnvironment(environb):
+            self.assertEqual(environb[self.goodKeyBytes], self.goodValueBytes)
+
+        return p.getResult().addCallback(gotEnvironment)
```

Comment:
I prefer to have all the test data inside the test, in this way it is easier to read... similar to what you did in test_UTF8StringInEnvironment

and maybe have a single test.

```suggestion
        os.environ['key-\N{snowman}'] = 'value-\N{sun}'
        os.environ['key-ascii''] = 'value-ascii'
        self.addCleanup(operator.delitem, os.environ, 'key-\N{snowman}')
        self.addCleanup(operator.delitem, os.environ, 'key-ascii')
     
        p = GetEnvironmentDictionary.run(reactor, [], os.environ)

        def gotEnvironment(environb):
           self.assertEqual(environb[b'key-ascii'], b'value-ascii')
            self.assertEqual(environb[b'key-\xe2\x98\x83'], b'value-\xe2\x98\x89')

        return p.getResult().addCallback(gotEnvironment)
```

Reply:
OK I did this